<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <title>Randomly</title>
  <meta name="description" content="">
  <meta name="author" content="">

  <!-- Mobile Specific Metas
  –––––––––––––––––––––––––––––––––––––––––––––––––– -->
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <!-- FONT
  <link href="//fonts.googleapis.com/css?family=Raleway:400,300,600" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="../Skeleton-2.0.4/css/normalize.css">
  <link rel="stylesheet" href="../Skeleton-2.0.4/css/skeleton.css">
  <link rel="icon" type="image/png" href="images/favicon.png">
  –––––––––––––––––––––––––––––––––––––––––––––––––– -->
  <link href="https://fonts.googleapis.com/css?family=Raleway:400,300,600" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com/css?family=Lora:400,400i,700" rel="stylesheet">
</head>

<body>
  <div class="container">
    <div class="row" style="margin-top: 5%">

      <div class="four columns">
        <h1><a href="/" style="font-weight: bold; text-decoration:none; color:black">Randomly</a></h1>
        <a href="/">
        <center>
        <img alt="Randomly" class="logo-width" src="https://s3.amazonaws.com/randomly-static/img/Randomly.png">
        </center>
        </a>
        <a class="button u-full-width" href="#intro">Overview of the Problem</a>
        <a class="button u-full-width" href="#idea">The Idea</a>
        <a class="button u-full-width" href="#theory">Random Matrix Theory in 1 Minute</a>
        <a class="button u-full-width" href="#moredetail">More Detail</a>
        <a class="button u-full-width" href="#examp">An Example</a>
        <a class="button u-full-width" href="#inaction">Randomly Interactive Demo</a>
        <a class="button u-full-width" href="#cite">Citation</a>
        <a class="button u-full-width" href="#refs">References</a>
        <!--
        <a class="button u-full-width" href="https://github.com/RabadanLab/randomly">
          <span>Github</span>
          <img alt="Randomly on Github" src="https://s3.amazonaws.com/randomly-static/img/github-mark.svg" style="width:25px"/>
        </a>
        -->
        <div class="row" style="padding-bottom: 5%"> </div>
        <div class="row" style="padding-bottom: 5%"> </div>
      </div> <!-- three columns -->

      <div class="eight columns">
        <h3>A Library for Denoising Single-Cell Data with Random Matrix Theory</h3>

        <p><strong>Randomly</strong> is a python package from the <a href="https://rabadan.c2b2.columbia.edu/">Rabadan Lab</a> for denoising single-cell data using <b>Random Matrix Theory</b>.</p>

        <p>Randomly will soon be freely available on github (we're just finalizing the code. <i>Stay tuned!)</i></p>

        <h4 id="intro">Overview of the Problem</h4>

        <ul>
          <li>Characterizing the different cell subtypes in <b>heterogenous populations</b> and their evolution plays a central role in understanding complex systems like <b>cancer</b> or <b>immune cells</b></li>
          <li><b>Single-cell</b> technologies<sup>1,2</sup> offer the opportunity to identify new cell types and cellular states</li>
          <li>There are biological and technical challenges that complicate the analysis of <b>single-cell data</b>:
            <ul>
              <li><u>Problem 1</u>: a lack of a complete quantitative understanding of different sorts of <b>noise</b></li>
              <li><u>Problem 2</u>: <b>Sparsity</b> of data due to low amounts of amplified genomic material</li>
            </ul>
          </li>
          <li>Currently, there are two main methods to address these challenges:
            <ol>
              <li><i>Imputation of missing values</i>: try to infer the “true” expression of data samples by modeling the underlying distributions<sup>3,4</sup></li>
              <li><i>Zero-inflated dimensionality reduction</i>: a modified-PCA method modeling the underlying distribution<sup>5,6</sup></li>
            </ol>
          </li>
          <li>However, these methods make assumptions about gene expression distributions that are not well justified and may generate artifacts</li>
          <li>To remedy this, Randomly develops a method for denoising single-cell data in a universal way:
            <ul>
              <li><u>Objective 1</u>: develop a statistical description of the noise in single-cell data that is insensitive to the specific details of gene expression in cells</li>
              <li><u>Objective 2</u>: analyze and control the artifacts induced by sparsity in single-cell data</li>
            </ul>
          </li>
        </ul>

        <h4 id="idea">The Idea</h4>

        <p>
        This problem is very similar to one addressed in nuclear physics in the 1950s and later in a variety of complex systems<sup>7</sup>.
        The statistics of these systems turns out to be described by <b>Random Matrix Theory</b>.
        </p>

        <p>We observed that single-cell data resembles the Wigner Surmise:</p>
        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/wignersurmise.png"></p>        

        <h4 id="theory">Random Matrix Theory in 1 Minute</h4>

        <ul>
          <li>For a given single-cell dataset <i>X</i> with <i>N</i> cells and <i>P</i> genes, we can define a Wishart matrix:
            <br><br>
            <div id="eqn1" class="centerme"></div>
            <br>
            All the properties of <i>W</i> are described by its eigenvalues <span id="eqn2"></span> and eigenvectors <span id="eqn3"></span> in the form:
            <br><br>
            <div id="eqn4" class="centerme"></div>
            <br>
            If X is a random matrix, then there are some universal statistical features:
            <ol>
              <li>Eigenvalue density follows the <b>Marchenko-Pastur</b> (MP) distribution</li>
              <li>The largest eigenvalue is described by the <b>Tracy-Widom</b> distribution</li>
              <li>Eigenvectors are <b>delocalized</b></li>
            </ol>
          </li>
          <li><b>Universality</b> means that the statistical features are independent of the gene expression distribution. They only depend on the finiteness of the first few moments<sup>8,9</sup></li>
        </ul>

        <h4 id="moredetail">More Detail</h4>

        <p>
        <b>Deviations from the universal</b> eigenvalue distribution (Marchenko-Pastur) predicted by Random Matrix Theory indicate the presence of a <b>signal</b> that can be further analyzed:
        </p>

        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/noisesig2.png"></p>

        <p>Equivalently, the appearance of localized eigenvectors indicates the presence of a signal:</p>
        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/localeig.png"></p>
        <p>In contrast, the delocalized eigenvectors correspond to noise described by a random matrix:</p>
        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/deloceig.png"></p>

        <p>
        The delocalized eigenvector component distribution corresponds to the maximum entropy probability density function (PDF), which means that there is no information.
        On the other hand, the localized eigenvector component distribution doesn't fit this PDF and thus contains information.
        </p>

        <p>However, there is a subtlety: <b>part of the signal is an artifact due to the sparsity of the data</b>.</p>

        <p>
        A way of capturing this artifact is to completely randomize the single-cell matrix.
        According to the theory explained above, all the eigenvectors should be delocalized and the eigenvalues should follow a Marchenko-Pastur distribution.
        However, the sparsity of the data modifies this predicted behavior. We call this artifact <b>sparsity induced localization</b>.
        In the following plot, we apply a Gaussianity test to identify the localized eigenvectors corresponding to this artifact:
        </p>
        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/eigdist.png"></p>

        <p>
        The take-home message is the following:
        <ul>
          <li>artifacts due to the sparsity of the data, including sparsity induced localization, account for 3% of the signal eigenvalues on average</li>
          <li>on average, the final 2% of eigenvalues could then be attributed to a true biological signal</li>
          <li>universality of random matrix theory guarantees the validity of the method regardless of the single-cell technology:</li>
        </ul>
        </p>
        <p><img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/bars.png"></p>

        <h4 id="examp">An Example</h4>

        <p>
        As an example, we consider single-cell transcriptomic data from a set of 6,573 peripheral blood mononuclear cells (PBMC)<sup>12</sup>.
        </p>

        <p>
        Randomly applies the universality features of Sparse Random Matrix Theory<sup>10,11</sup> to separate noise and signal in single-cell data.
        The Randomly analysis consists of three main steps:
        </p>

        <ul>
          <li>
          Separation of noise and signal using universality of MP and Tracy-Widom distributions:
          <img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/mp.png">
          </li>

          <li>
          Removal of the sparse signal in the dataset by studying the sparsity induced localization
          </li>

          <li>
          Selection of the genes which are most responsible for signal.
          To accomplish this, we project the genes into signal and noise components, and then study the sample variance via a chi-squared test.
          From that, we establish a false discovery rate to identify genes that generate signal:
          <img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/geneprojection.png">
          </li>
        </ul>

        <p>
        We have projected out the noise and selected the top 1,000 genes most responsible for signal according to the step just described.
        We use these projected genes to do a standard hierarchical clustering and visualize it using t-SNE.
        We compare our clustering with the cell labels provided in references [12] and [13].
        <img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/clustering.png">
        </p>

        <p>
        We compare the performance of Randomly (RMT) with other algorithms in terms of cell-phenotype cluster resolution.
        For completeness, we also compare with the raw data and with a selection of the top 300 genes based on highest variance (300 mvg).
        Provided known cell-phenotypes from [13], the mean silhouette score quantifies the cluster resolution.
        The comparison is performed as a function of the reduced-space number of dimensions (number of principal components).
        <br>
        <br>
        <img class="imgstyle1" src="https://s3.amazonaws.com/randomly-static/img/comparison.png">
        </p>

        <h4 id="inaction">Randomly Interactive Demo</h4>

        <p>
        The following demo shows a t-SNE visualization of phenotypes of mouse-cortex cells provided in [14].
        </p>

        <p>
          <div id="attachsvg"></div>
          <div id="chart"></div>
          <div id="menu">
            <div>
              Perplexity: <span id="perpVal"></span>
            </div>
            <br>
            <div class="slider" id="perpSlider"></div>
            <br>
            <div>
              lambda range: <span id="lambda1Val"></span> - <span id="lambda2Val"></span>
            </div>
            <br>
            <div class="slider" id="lambdaSlider"></div>
            <br>
            <button type="button" onclick="demo_l1()">auto l1</button>
            <button type="button" onclick="demo_l2()">auto l2</button>
            <button type="button" onclick="interval_stop()">stop</button>
          </div>
        </p>

        <h4 id="cite">Citation</h4>

        <p>
        For now, you can cite the Randomly preprint as:
        </p>

        </p>
        <b>Quasi-universality in single-cell sequencing data</b>.
        <br>
        Luis Aparicio, Mykola Bordyuh, Andrew J. Blumberg, Raul Rabadan.
        <br>
        <a href="https://www.biorxiv.org/content/early/2018/10/05/426239"><i>biorXiv preprint</i> doi.org/10.1101/426239 (2018)</a>.
        </p>

        <h4 id="refs">References</h4>
        <ol>
          <li>L. Bintu et al., Dynamics of epigenetic regulation at the single-cell level. <i>Science</i> <b>351</b>, 720-724 (2016)</li>
          <li>J. Cao et al., Comprehensive single-cell transcriptional profiling of a multicellular organism. <i>Science</i> <b>357</b>, 661-667 (2017)</li>
          <li>D. van Dijk et al., Recovering Gene Interactions from Single-Cell Data Using Data Diffusion. <i>Cell</i> <b>174</b>, 716-+ (2018)</li>
          <li>W. V. Li, J. Y. J. Li, An accurate and robust imputation method scImpute for single-cell RNA-seq data. <i>Nat Commun</i> <b>9</b>,  (2018)</li>
          <li>E. Pierson, C. Yau, ZIFA: Dimensionality reduction for zero-inflated single-cell gene expression analysis. <i>Genome Biol</i> <b>16</b>,  (2015)</li>
          <li>D. Risso, F. Perraudeau, S. Gribkova, S. Dudoit, J. P. Vert, A general and flexible method for signal extraction from single-cell RNA-seq data. <i>Nat Commun</i> <b>9</b>,  (2018)</li>
          <li>F. J. Dyson, Statistical Theory of Energy Levels of Complex Systems .1. <i>J Math Phys</i> <b>3</b>, 140-& (1962)</li>
          <li>T. Tao, V. Vu, Random Matrices: Universality of Local Eigenvalue Statistics up to the Edge. <i>Commun Math Phys</i> <b>298</b>, 549-572 (2010)</li>
          <li>N. S. Pillai, J. Yin, Universality of Covariance Matrices. <i>Ann Appl Probab</i> <b>24</b>, 935-1001 (2014)</li>
          <li>Y. V. Fyodorov, A. D. Mirlin, Localization in ensemble of sparse random matrices. <i>Phys Rev Lett</i> <b>67</b>, 2049-2052 (1991)</li>
          <li>S. N. Evangelou, E. N. Economou, Spectral Density Singularities, Level Statistics, and Localization in a Sparse Random Matrix Ensemble. <i>Physical Review Letters</i> <b>68</b>, 361-364 (1992)</li>
          <li>H. M. Kang et al., Multiplexed droplet single-cell RNA-sequencing using natural genetic variation. <i>Nat Biotechnol</i> <b>36</b>, 89-94 (2018)</li>
          <li>A. Butler, P. Hoffman, P. Smibert, E. Papalexi, R. Satija, Integrating single-cell transcriptomic data across different conditions, technologies, and species. <i>Nature Biotechnology</i> <b>36</b>, 411-+ (2018)</li>
          <li>Amit Zeisel, Ana B. Muñoz-Manchado, Simone Codeluppi et al. Cell types in the mouse cortex and hippocampus revealed by single-cell rna-seq. <i>Science</i>, <b>347(6226)</b> 1138–1142, 2015</li>
        </ol>

      </div> <!-- nine columns -->

    </div> <!-- row -->

    <div class="row" style="padding-bottom: 25%">
      <a href="https://rabadan.c2b2.columbia.edu/">Rabadan Lab</a> &copy; 2018
    </div>
  </div> <!-- container -->
</body>

</html>
